### ISSUE TITLE
Type style fails to recognize predefined types by their metadata names

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#20244]
**Version Used**: 15.2

:link: Originally reported by Josh Hufford: [Suggesting both 'var' and explicit type.](https://developercommunity.visualstudio.com/content/problem/68755/suggesting-both-var-and-explicit-type.html)

**Steps to Reproduce**:

1. Set the style for **For built-in types** to **Prefer explicit type**
1. Set the style for **When variable type is apparent** to **Prefer 'var'**
1. Use the following code

    ```csharp
    String test = new String(' ', 4);
    ```

**Expected Behavior**:

No warning is reported, because `String` is the built-in type `string`.

**Actual Behavior**:

Warning IDE0007 is reported.

**Comment by MattGertz** [comment]
Mondo escrow triage:  moving out.
